core/src/main/scala/org/apache/spark/h2o/converters/H2ODataFrame.scala

```diff
@@ -92,31 +94,25 @@ class H2ODataFrame[T <: water.fvec.Frame](@transient val frame: T,
         expectedTypes,
         partIndex)
 
+      /*a sequence of converters, per column*/
+      private val columnConverters = filteredTypes map converterCtx.get
+
+
+      private def readRow: InternalRow = {
+          val optionalData: Seq[Option[Any]] =
+            columnConverters.zipWithIndex map { case (c, i) => c(selectedColumnIndices(i)) }
```

Comment:
Instead of i `idx`, instead of c `convertPerCol`, just to make it more readable ?